### ISSUE TITLE
Look into libsysstat/libstatgrab situation

### ISSUE THREAD

**Issue by jleclanche** [lxqt/lxqt#704]
- Figure out what can be dropped in favour of KF5::Solid
- Figure out what functionality, if any, is still needed in our current libraries
- If there's any remaining, see whether we can simply do without that functionality, as their use is limited to lxqt-panel.

**Comment by paulolieuthier** [comment]
We actually depend on lm_sensors as well, in plugin-sensors.

**Comment by paulolieuthier** [comment]
http://sourceforge.net/p/lxde/mailman/message/31583019/

**Comment by paulolieuthier** [comment]
This is the situation:
* The sensors plugin depends on lm_sensors.
* The network monitor plugin depends on libstatgrab.
* The sysstat plugin depends on libsysstat.

I think we should choose libsysstat or libstatgrab to provide information for both the sysstat and network plugins. I don't think these two libraries can do what lm_sensors does.

@kuzmas what do you think? I saw your email (linked above), but it's been been two years since that and libstatgrab seems to have matured.

**Comment by kuzmas** [comment]
@paulolieuthier sorry, I haven't been following libstatgrab or lm_sensors features for the last two years.
What I can say for sure is that libsysstat can produce _synchronised_ graphs of cpu, memory, swap and network load. Can lm_sensors or libstatgrab do this? If yes - ditch libsysstat.

tl;dr: if a plugin+library can do something unique then it's better to keep them.

**Comment by jleclanche** [comment]
@kuzmas agreed, if there's no alternative, we can keep maintaining it. But if it's available in a kf5 or some such, I'd rather we drop it. Nobody is working on it so it just goes unmaintained.

**Comment by paulolieuthier** [comment]
There is yet another plugin, cpu load, that depends on libstatgrab. I tried porting the network monitor plugin to libsysstat, but the library lacks a fundamental feature: show the global statistics. It only signals the change, with no information about the accumulated resource usage. To put it more clearly: one can set an interval and get updates of how many bytes were transferred and received in each interval. But one can't get information about how much was transmitted/received since computer startup.

That information is provided easily by libstatgrab. Sure it can be added to libsysstat with no much effort. And it's nice a Qt-based library around. But it means maintenance cost for us. Besides, libstatgrab is very portable and have information about processes, users, disks, filesystems etc.

I'll start porting the sysstat plugin to libstatgrab.

**Comment by jleclanche** [comment]
> To put it more clearly: one can set an interval and get updates of how many bytes were transferred and received in each interval.

1. Is that possible?
2. Is that even something we want? I think it's OK to display stats since lxqt startup.

**Comment by paulolieuthier** [comment]
> Is that possible?

Yes, both libstatgrab and libsysstat do that by reading `/proc/net/dev`. It's simple. The problem is that libsysstat only gives the additions on the previous reads.

> Is that even something we want? I think it's OK to display stats since lxqt startup.

Yes, we want it. The network monitor plugin can switch its icon depending on whether it has or hasn't be transmitted/received bytes in the previous interval.

**Comment by jleclanche** [comment]
> Yes, we want it. The network monitor plugin can switch its icon depending on whether it has or hasn't be transmitted/received bytes in the previous interval.

That's not a good enough reason tbh

**Comment by paulolieuthier** [comment]
> That's not a good enough reason tbh

What do you mean? Isn't this the core feature of the network monitor plugin (besides showing the stats since startup)?

**Comment by paulolieuthier** [comment]
I had some more time and looked further into it. There is one more reason to deprecate libstatgrab: it only supports Linux. libstatgrab, as I said, is very portable. It supports the BSDs as well. So I started the port. However, libstatgrab, **AFAIK**, can't give the stats of individual cores of a multi-code CPU. libsysstat can. It means the port would remove features from the plugin. I'm not sure what to do. Maybe we should postpone this issue.

**Comment by paulolieuthier** [comment]
libstatgrab also doesn't know about the CPU frequency.

**Comment by palinek** [comment]
I don't use the stat plugin but IMO such plugin should be aware of cpu cores and their requencies.

**Comment by agaida** [comment]
imho we have two choices - deprecate libsysstat or make sure that the
plugins aren't built for not supported platforms - that should be easy
with cmake. Problem solved :) 

On the other hand - do we need the statistic tools at all? a overall
stat would be good - maybe linked to a more detailed tool like qps.

But wait - qps in its current state runs only on linux and maybe
solaris so it should be improved. Another idea is to have a look at the
code from lxde. Maybe this could help.


-- 
Alf Gaida
BDBF C688 EFAD BA89 5A9F  464B CD28 0A0B 4D72 827C

**Comment by jsm222** [comment]
FYI Olivier Duchateau and my self, have ported libsysstat to FreeBSD, with some minor differences, e.g network is net io on all interfaces (e.g no source) and you cannot use the frequency for the cpu. see https://github.com/jsm222/lxqt-freebsd/tree/master/sysutils/libsysstat. libstatgrab seems to be planning per cpu stats as well https://github.com/i-scream/libstatgrab/issues/13 but they do not seem to update too frequently at the moment.

**Comment by forraskod** [comment]
Hello,

I really would like libsysstat will be able to print different values in numbers not only diagrams. Do you plan to do it?
